### DISCUSSION TITLE
Command 'ps' required by nextflow to collect task metrics cannot be found

### DISCUSSION THREAD

**Discussion by pouya-codes** [nextflow-io/nextflow#3081] [Q&A] [answered]
## Bug report 

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior

I am training to get the log of my job execution by using the -with-report flag, but I'm getting the following error.
I am using slurm executor to submit the jobs, and all the excitation nodes have ps installed.

### Steps to reproduce the problem

(Provide a test case that reproduces the problem either with a self-contained script or GitHub repository)

### Program output 

Command exit status:
  1

Command output:
  (empty)

Command error:
  Command 'ps' required by nextflow to collect task metrics cannot be found

Work dir:
  /projects/ovcare/classification/cPathPortal/works/pipeline/91/d436a25b21b9ccc1ca7b7c2fc6ea6b

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`

### Environment 

* Nextflow version: [22.05.0-edge build 5704] 
* Java version: [11.0.15]
* Operating system: [Linux]
* Bash version: 4.4.20

### Additional context

I am using the head node of the slurm cluster to execute my jobs on the executor nodes. All the processes execute by using a singularity environment.

**Comment by tbugfinder** [comment, marked as answer]
Typically the singularity container lacks the ps command. Did you check it being installed?

**Reply by pouya-codes** [reply]
Installing ps on the container solve the error thanks :)

**Reply by IvanWoo22** [reply]
Hi, I met the same error when running nf-core's workflow. When I independently run `command -v ps`, I got a `/usr/bin/ps` as result. I think `ps` is installed on our lsf sys, but I don't know how to check it in the singularity container. Could you give me some help?

**Reply by mribeirodantas** [reply]
You should be able to interactively enter your singularity container with `singularity shell container.sif`

**Reply by TKNsama** [reply]
Hi, I met the same error too.
I checked the `ps` command in `singularity shell .sif`. the ps is presented. but still nextflow can not found the ps command.
```
Command error:
  .command.run: line 147: /dev/null: Permission denied
  Command 'ps' required by nextflow to collect task metrics cannot be found
  cp: '.command.out' and '.command.out' are the same file
  cp: '.command.err' and '.command.err' are the same file
  cp: '.command.trace' and '.command.trace' are the same file
```
could you please give me some help?
